decisions/DR007-Hierarchical-or-compositional-traits-for-specifications.md

```diff
@@ -0,0 +1,361 @@
+    _type = "blob"
+    mimeType = Specification.TypedProperty(str, doc="")
+
+
+class ContainerTrait(specifications.EntityTrait):
+    _type = "container"
+
+
+class ImageTrait(specifications.EntityTrait):
+    _type = "image"
+
+    channelNames = Specification.TypedProperty(list, doc="")
+    colorSpace = Specification.TypedProperty(str, doc="")
+
+
+class RasterTrait(specifications.EntityTrait):
+    _type = "raster"
+
+    aspectRatio = Specification.TypedProperty(list, doc="")
+    resolution = Specification.TypedProperty(list, doc="")  # E.g. [“1920x1080”, “420x420”]
+
+
+class GeometryTrait(specifications.EntityTrait):
+    _type = "geometry"
+
+    # startFrame==endFrame means "not animated"
+    startFrame = Specification.TypedProperty(float, doc="")
+    endFrame = Specification.TypedProperty(float, doc="")
+
+
+class EditorialTrait(specifications.EntityTrait):
+    _type = "editorial"
+
+    startFrame = Specification.TypedProperty(int, doc="")
+    endFrame = Specification.TypedProperty(int, doc="")
+
+
+class ShotTrait(specifications.EntityTrait):
+    _type = "shot"
+
+
+class DocumentTrait(specifications.EntityTrait):
+    _type = "document"
+
+
+class NukeScriptTrait(specifications.EntityTrait):
+    _type = "nukescript"
+
+    usesNukeXNodes = Specification.TypedProperty(bool, doc="")
+
+
+class KatanaProjectTrait(specifications.EntityTrait):
+    _type = "katanaproject"
+
+
+class CacheTrait(specifications.EntityTrait):
+    _type = "cache"
+
+
+class ImageSpecification(BlobTrait, ImageTrait):
+    _type = ".".join(sorted((BlobTrait._type, ImageTrait._type)))
+
+
+class RasterImageSpecification(BlobTrait, ImageTrait, RasterTrait):
+    """
+    Raster image, as opposed to e.g. SVG.
+    """
+    _type = ".".join(sorted((BlobTrait._type, ImageTrait._type, RasterTrait._type)))
+
+
+class ImageBundleSpecification(ContainerTrait, ImageTrait):
+    _type = ".".join(sorted((ContainerTrait._type, ImageTrait._type)))
+
+
+class SceneCacheSpecification(GeometryTrait, CacheTrait, BlobTrait):
+    _type = ".".join(sorted((GeometryTrait._type, CacheTrait._type)))
+
+
+class EditorialSpecification(ContainerTrait, EditorialTrait):
+    _type = ".".join(sorted((ContainerTrait._type, EditorialTrait._type)))
+
+
+class ShotSpecification(ContainerTrait, EditorialTrait, ShotTrait):
+    _type = ".".join(sorted((ContainerTrait._type, EditorialTrait._type, ShotTrait._type)))
+
+
+class DocumentSpecification(BlobTrait, DocumentTrait):
+    _type = ".".join(sorted((BlobTrait._type, DocumentTrait._type)))
+
+
+class NukeScriptSpecification(BlobTrait, DocumentTrait, NukeScriptTrait):
+    _type = ".".join(sorted((BlobTrait._type, DocumentTrait._type, NukeScriptTrait._type)))
+
+
+class KatanaProjectSpecification(BlobTrait, DocumentTrait, KatanaProjectTrait):
+    _type = ".".join(sorted((BlobTrait._type, DocumentTrait._type, KatanaProjectTrait._type)))
+```
+
+### An example of using these specifications
+
+#### Checking management policies
```

Comment:
[consider] Is it worth noting that this and the ones below is identical for option 1 and option 2?